final-paper/final-paper.org

```diff
@@ -61,22 +60,23 @@ Since vectors with the greatest similarity may not satisfy the predicate, this a
-An oracle partition index is more performant compared to a base ACORN index because the search space is guaranteed to contain only vectors that satisfy the predicate, and thus no predicate comparison must occur during the search itself.
-The reason ACORN provides a /predicate-agnostic/ index is because in many systems, it is infeasible to construct an index for every possible predicate, and thus a general-purpose index performs well across a diverse array of queries.
+An oracle partition index is more performant compared to a base ACORN index because the search space is guaranteed to contain only vectors that satisfy the predicate, and thus no further filtering needs to occur before, during, or after the search itself.
+In many cases, it is infeasible to construct an oracle index for every possible predicate, and thus a general-purpose index such as ACORN is more desirable, as it performs reasonably well across a diverse array of queries and is much more space efficient.
 
-However, access patterns are sometimes not uniformly distributed.
-Consider, for example, queries about football teams during Superbowl or states during election night.
-In certain scenarios, a general-purpose predicate-agnostic index can be supplemented by carefully chosen supplemental indices constructed using frequently occurring predicates.
-This is the goal of OAK.
+There are, however, many query distributions could be get significant and meaningful performance improvement if an oracle index were to exist that could be searched rather than an index that also contains vectors that don't match the search predicate.
+The greater the scale of the database and the volume of queries, the more critical such a performance improvement could be. 
+Consider, for example, queries about football teams during the Superbowl, or about states during election night.
+In scenarios like this, a general-purpose predicate-agnostic index can be supplemented by carefully chosen supplemental indices that are constructed so as to effectively serve some frequently occurring search predicates.
+Intelligently routing to supplemental indexes in such cases is the goal of OAK.
 
 * Background and Related Work
 
-The underlying data structure of both ACORN and OAK is Hierarchical Navigable Small Worlds (HNSW)[cite:https://arxiv.org/abs/1603.09320], a hierarchical, tree-link structure where each node of the tree represents a set of vectors.
-More specifically, it is a /proximity graph/, in which two vertices are linked based on proximity.
+The underlying data structure that enables vector similiarity search in both ACORN and OAK is Hierarchical Navigable Small Worlds (HNSW) [cite:@malkovEfficientRobustApproximate2018], an index that makes use of a hierarchical, tree-link structure during search to more quickly work through the plausible search space while retaining reasonable recall.
+More specifically, HNSW leverages a /proximity graph/, in which two vertices are linked based on proximity.
```

Comment:
@csirianni what are you referring to here when talking about HNSW's proximity graph?

Reply:
https://www.pinecone.io/learn/series/faiss/hnsw/

> HNSW slots into the graph category. More specifically, it is a proximity graph, in which two vertices are linked based on their proximity (closer vertices are linked) — often defined in Euclidean distance.